### DISCUSSION TITLE
Android->iOS color format is not transposed

### DISCUSSION THREAD

**Discussion by ClarkBattle-McFee** [SwiftGen/SwiftGen#868] [Q&A] [answered]
I have a question about the colors. I have an android colors.xml that I would like to use to generate a Colors.swift. SwiftGen works in that it generates the file, however, android color format is ARGB, whils iOS' color format is RGBA. SwiftGen does not do the translation. 

    <color name="button_primary_active_border">#ff000000</color>

becomes

    public static let buttonPrimaryActiveBorder = ColorEnum(rgbaValue: 0xff000000)

when it should be

    public static let buttonPrimaryActiveBorder = ColorEnum(rgbaValue: 0x000000ff)

Is there a way to fix this?

**Comment by ClarkBattle-McFee** [comment, marked as answer]
I just found this https://github.com/SwiftGen/SwiftGen/blob/stable/Documentation/Parsers/colors.md#customization
So it looks like there is support for colorFormat: argb. I'm not clear on where to put that though. My swiftgen.yml looks like this

    colors:
      inputs: ${SRCROOT}/resources/colors.xml
      outputs:
          - templateName: swift5
            output: Colors.swift
            params:
                publicAccess: true
                enumName: ColorEnum

Where should I put the colorFormat: argb?

**Reply by ClarkBattle-McFee** [reply]
I figured it out. 

    colors:
      inputs: ${SRCROOT}/resources/colors.xml
      options:
          colorFormat: argb
      outputs:
          - templateName: swift5
            output: Colors.swift
            params:
                publicAccess: true
                enumName: ColorEnum

**Reply by djbe** [reply]
Glad you were able to find the solution 😄
